ipn/ipnserver/server.go

```diff
@@ -613,9 +613,18 @@ func Run(ctx context.Context, logf logger.Logf, logid string, getEngine func() (
 
 	var store ipn.StateStore
 	if opts.StatePath != "" {
-		store, err = ipn.NewFileStore(opts.StatePath)
-		if err != nil {
-			return fmt.Errorf("ipn.NewFileStore(%q): %v", opts.StatePath, err)
+		const kubePrefix = "kube:"
+		switch {
+		case strings.HasPrefix(opts.StatePath, kubePrefix):
+			store, err = ipn.NewKubeStore(opts.StatePath[len(kubePrefix):])
+			if err != nil {
+				return fmt.Errorf("ipn.NewKubeStore(%q): %v", opts.StatePath, err)
```

Comment:
technically ipn.NewKubeStore was called with `dingus`, not `opts.StatePath`.

another reason for a variable.